### PR TITLE
feat: introduce docker-sync

### PR DESCRIPTION
**Pull request by Atala**
Introducing [docker-sync](http://docker-sync.io/). It is kind of shitty becaus it's a hack on top of docker but the dev experience is aaaaamzazing with it (response time ~200ms)

@alexsegura can you test it on your machine?

### CODE REVIEW

File: docker/php/Dockerfile [hunk 1]
```diff
@@ -72,6 +72,6 @@ COPY docker/php/start.sh /usr/local/bin/docker-app-start
 RUN chmod +x /usr/local/bin/docker-app-start
 
 # Needed to fix permissions error
-USER 1000
+USER 100
```
Comment (alexsegura) [inline docker/php/Dockerfile]: Doesn't this break standard `docker-compose`?

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+services:
+
+  webpack:
+    volumes:
+      - 'node_modules:/srv/coopcycle/node_modules'
+      - 'webpack-sync:/srv/coopcycle:nocopy'
+
+  php:
+    volumes:
+      - 'node_modules:/srv/coopcycle/node_modules'
```
Comment (alexsegura) [inline docker-compose-dev.yml]: Why `node_modules`?

**Comment by alexsegura** [comment]
It works, and it feels a bit faster, yes. 
I never had such performance problems on my machine. 

Side note: maybe we should specify in the `README` that it is needed to run `docker-compose build` when containers have changed?

:shipit:

File: docker/php/Dockerfile (hunk 1 above)
Comment (alexsegura) [reply docker/php/Dockerfile]: Looks like `docker-compose` still works. 
I don't know why but it works 🤔

**Comment by Atala** [comment]
I think even better would be sthg like [this](https://gist.github.com/mariusGundersen/8a4383f74a46ababc1de) -> run container build if Dockerfile as changed.

Should work with migrations, package.json, aso. too

File: docker-compose-dev.yml (hunk 2 above)
Comment (Atala) [reply docker-compose-dev.yml]: That is a useless line.

File: docker/php/Dockerfile (hunk 1 above)
Comment (Atala) [reply docker/php/Dockerfile]: I tested it work on my side.

The main issue I had with `docker-sync` is permission issues, it does not seem to work with an user ID of `1000` (I don't know why)

**Pull request merged**